glommio/src/io/bulk_io.rs

```diff
@@ -14,6 +14,51 @@ use std::{
     rc::Rc,
 };
 
+/// Set a limit to the size of merged IO requests.
+#[derive(Debug)]
+pub enum MergedBufferLimit {
+    /// Disables request coalescing
+    NoMerging,
+
+    /// Sets the limit to the maximum the kernel allows for the underlying
+    /// device without breaking down the request into smaller ones
+    /// (/sys/block/.../queue/max_sectors_kb)
+    DeviceMaxSingleRequest,
+
+    /// Sets the limit to the maximum the kernel allows for the underlying
+    /// device (/sys/block/.../queue/max_segment_size)
+    DeviceMax,
```

Comment:
The old behavior had "unlimited" but when I discovered that the kernel actually has a hard limit I introduced this instead to prevent the stream from failing.  Esp with in-memory files, I think such fat requests like this _could_ happen. 

PS: It is the same limit on my side and on an i3.

Reply:
I am not sure what this buys us. It's a property we don't fully understand, that will essentially be unlimited until it is zero in someone's system for whatever reason you didn't anticipate.

Reply:
Do you think I should remove the option and limit values we are given under the hood? Or just not use `max_segment_size`?